File: apps/api/app/schemas/audio.py
from uuid import UUID
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class AudioResponse(BaseModel):
    node_id: UUID
    language: str
    code_mix_ratio: float
    speaker: str
    audio_url: str
    duration_sec: Optional[float] = None
    file_size: Optional[int] = None
    is_cached: bool = True
    expires_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True


class AudioGeneratingResponse(BaseModel):
    node_id: UUID
    language: str
    status: str = "generating"
    estimated_wait_sec: int = 5
    retry_after: int = 5

File: apps/web/my-app/types/audio.ts
export interface AudioResponse {
  node_id: string;
  language: string;
  code_mix_ratio: number;
  speaker: string;
  audio_url: string;
  duration_sec?: number;
  file_size?: number;
  is_cached: boolean;
  expires_at?: string;
}

export interface AudioGeneratingResponse {
  node_id: string;
  language: string;
  code_mix_ratio: number;
  speaker: string;
  audio_url: string;
  status: 'generating';
  estimated_wait_sec: number;
  retry_after: number;
}

export interface AudioParams {
  language: string;
  speaker?: string;
  code_mix?: number;
}
